From a30bf9267901e197285b4882bf24bb22cdcdf3ef Mon Sep 17 00:00:00 2001
From: Chenxi Mao <chenxi.mao@suse.com>
Date: Thu, 17 Nov 2022 09:37:47 +0800
Subject: [PATCH 1/1] Support make rpm-pkg on x86_64 and aarch64 platform

Patch-mainline: Never
Modified-by-SEL: N/A
References: bsn#371

1. Add Module.supported into SRPM to mark moudles as supported
2. Leverage mkinitrd to create initramfs image and update grub menu

Signed-off-by: Chenxi Mao <chenxi.mao@suse.com>
---
 scripts/Makefile.package |    2 +-
 scripts/package/mkspec   |   16 +++++++++-------
 2 files changed, 10 insertions(+), 8 deletions(-)

--- a/scripts/Makefile.package
+++ b/scripts/Makefile.package
@@ -29,7 +29,7 @@ KDEB_SOURCENAME ?= linux-$(KERNELRELEASE
 KBUILD_PKG_ROOTCMD ?="fakeroot -u"
 export KDEB_SOURCENAME
 # Include only those top-level files that are needed by make, plus the GPL copy
-TAR_CONTENT := $(KBUILD_ALLDIRS) .config .scmversion Makefile \
+TAR_CONTENT := $(KBUILD_ALLDIRS) .config .scmversion Makefile Module.supported \
                Kbuild Kconfig COPYING $(wildcard localversion*)
 MKSPEC     := $(srctree)/scripts/package/mkspec
 
--- a/scripts/package/mkspec
+++ b/scripts/package/mkspec
@@ -88,8 +88,12 @@ $S
 	cp \$($MAKE -s image_name) %{buildroot}/boot/efi/vmlinuz-$KERNELRELEASE
 	ln -s efi/vmlinuz-$KERNELRELEASE %{buildroot}/boot/
 	%else
+	%ifarch aarch64
+	cp \$($MAKE -s image_name) %{buildroot}/boot/Image-$KERNELRELEASE
+	%else
 	cp \$($MAKE -s image_name) %{buildroot}/boot/vmlinuz-$KERNELRELEASE
 	%endif
+	%endif
 $M	$MAKE %{?_smp_mflags} INSTALL_MOD_PATH=%{buildroot} modules_install
 	$MAKE %{?_smp_mflags} INSTALL_HDR_PATH=%{buildroot}/usr headers_install
 	cp System.map %{buildroot}/boot/System.map-$KERNELRELEASE
@@ -108,13 +112,11 @@ $S$M	ln -sf /usr/src/kernels/$KERNELRELE
 	rm -rf %{buildroot}
 
 	%post
-	if [ -x /sbin/installkernel -a -r /boot/vmlinuz-$KERNELRELEASE -a -r /boot/System.map-$KERNELRELEASE ]; then
-	cp /boot/vmlinuz-$KERNELRELEASE /boot/.vmlinuz-$KERNELRELEASE-rpm
-	cp /boot/System.map-$KERNELRELEASE /boot/.System.map-$KERNELRELEASE-rpm
-	rm -f /boot/vmlinuz-$KERNELRELEASE /boot/System.map-$KERNELRELEASE
-	/sbin/installkernel $KERNELRELEASE /boot/.vmlinuz-$KERNELRELEASE-rpm /boot/.System.map-$KERNELRELEASE-rpm
-	rm -f /boot/.vmlinuz-$KERNELRELEASE-rpm /boot/.System.map-$KERNELRELEASE-rpm
-	fi
+	%ifarch aarch64
+	mkinitrd -k Image-$KERNELRELEASE -i initramfs-$KERNELRELEASE.img
+	%else
+	mkinitrd -k vmlinuz-$KERNELRELEASE -i initramfs-$KERNELRELEASE.img
+	%endif
 
 	%preun
 	if [ -x /sbin/new-kernel-pkg ]; then
